File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(sum, numbers, i, target):
    if i == len(numbers):
        if sum == target:
            return 1
        else:
            return 0

    return dfs(sum + numbers[i], numbers, i+1, target)+ dfs(sum - numbers[i], numbers, i+1, target)

def solution(numbers, target):
    return dfs(0, numbers, 0, target)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    
    static int dfs(int[] numbers, int sum, int idx, int target){
        if(idx == numbers.length) {
            if(target == sum)
                return 1;
            else
                return 0;
        }
        
        return dfs(numbers, sum + numbers[idx], idx + 1, target) + dfs(numbers, sum - numbers[idx], idx + 1, target);
    }
    
    public int solution(int[] numbers, int target) {
        int answer = 0;
        
        answer = dfs(numbers, 0, 0, target);
        
        return answer;
    }
}